include/deal.II/grid/tria.h

```diff
@@ -3513,6 +3527,13 @@ class Triangulation : public Subscriptor
    */
   std::vector<ReferenceCell> reference_cells;
 
+  /**
+   * Constructor to be used only by parallel Triangulation classes.
+   */
+  Triangulation(const MPI_Comm &    mpi_communicator,
```

Comment:
I think this is the wrong approach. The class is serial, it should not have a constructor that takes a communicator, nor should it store a communicator.